src/action/ActionInterface.ts

```diff
@@ -0,0 +1,72 @@
+import type { BaseAction, ActionId } from "../types";
+import { ActionImpl } from "./ActionImpl";
+
+export class ActionInterface {
+  actions: Record<ActionId, ActionImpl> = {};
```

Comment:
This interface improves the previous naive logic we had for adding/removing actions. This way is more testable and extracts logic out of our React component.